src/adapter/src/catalog.rs

```diff
@@ -363,6 +368,49 @@ impl Catalog {
         }
         policies
     }
+
+    /// Return a set of [`GlobalId`]s for items that need to have their cache entries invalidated
+    /// as a result of creating new indexes on the items in `ons`.
+    pub(crate) fn invalidate_for_index(
+        &self,
+        ons: impl Iterator<Item = GlobalId>,
+    ) -> BTreeSet<GlobalId> {
+        let mut dependencies = BTreeSet::new();
+        let mut queue = VecDeque::new();
+        let mut seen = HashSet::new();
+        for on in ons {
+            dependencies.insert(on);
+            seen.insert(on);
+            let entry = self.get_entry(&on);
+            let uses = entry.uses();
+            queue.extend(uses.clone());
+            seen.extend(uses);
```

Comment:
My brain might be too smooth already, but I don't get how this one works. When we start out, `queue == seen`, so won't the check on `seen.insert(cur)` always be false, so we don't do anything?

Reply:
Hmm, I believe that it is my brain that is the smooth one. This line shouldn't be here. I've pushed a fix.

Reply:
✊